pymeasure/instruments/gwinstek/gwinstekGPD2303S.py

```diff
@@ -0,0 +1,230 @@
+    a baud rate of 9600. \n
+
+    the GPD-230s series only has two channels.
+
+    Supported commands (as per the manual):
+
+      - VSET<CH>:<volts>
+      - VSET<CH>?
+      - ISET<CH>:<amps>
+      - ISET<CH>?
+
+      - VOUT<CH>?
+      - IOUT<CH>?
+      - TRACK<NR1> (0=Independent, 1=Series, 2=Parallel)
+
+      - OUT<BOOLEAN> (0=Off, 1=On)
+      - LOCAL / REMOTE
+
+      - BEEP<BOOLEAN>
+      - SAV<NR1> / RCL<NR1> (memory save/recall)
+
+      - BAUD<NR1> (0:115200, 1:57600, 2:9600)
+      - *IDN? / STATUS? / ERR? / HELP?
+
+      - SAV/RCL do not affect output state; OUT may be turned off on save/recall as safety
+
+    .. code-block:: python
+
+    ps.output_enabled = 1
+    ps.channel_1.voltage_limit=5
+    ps.channel_1.current_limit=1
+    print(ps.error)
+    print(ps.status)
+    print(ps.channel_1.voltage)
+    print(ps.channel_1.current)\n
+
+    Known issue:
+    - When the instrument throws an error, it is best to flush the read buffer
+    `ps.adapter.flush_read_buffer()`
+    """
+
+    channel_1 = Instrument.ChannelCreator(GWInstekGPD230SChannel, 1)
+    channel_2 = Instrument.ChannelCreator(GWInstekGPD230SChannel, 2)
+
+    def __init__(
+        self,
+        adapter,
+        name="GW Instek GPD-2303S DC Power Supply",
+        read_termination="\n",
+        baud_rate=9600,
+        **kwargs,
+    ):
+        super().__init__(
+            adapter,
+            name,
+            read_termination=read_termination,
+            baud_rate=baud_rate,
+            includeSCPI=False,
+            **kwargs,
+        )
+
+    @staticmethod
+    def decode_status_byte(status_byte: int) -> dict:
+        """Decode the status byte according to manual."""
+        if type(status_byte) is not int:
+            raise (
+                ValueError((f"status_byte {status_byte} is not type int but {type(status_byte)}"))
+            )
+        return_dict = {}
+        return_dict["raw"] = bin(status_byte)
+        for register in GWInstekGPD2303SStatusRegister:
+            name = register.name
+            flag = register.value["flag"]
+            states = register.value["states"]
+
+            trailing_zeors = (flag & -flag).bit_length() - 1
+            state = (flag & status_byte) >> trailing_zeors
+
+            return_dict[name] = states.get(state, "Error")
+
+        return return_dict
+
+    @property
+    def status(self) -> dict:
+        """Get the status byte and and decode it."""
+        status_byte = int(self.ask("STATUS?").strip(), 2)
+        return self.decode_status_byte(status_byte)
+
+    # remote = Instrument.setting(
+    #     "REMOTE",
+    #     """ Exits local mode and sets the instrument to remote mode""",
+    # )
+
+    tracking_mode = Instrument.setting(
+        "TRACK%d",
+        """Set the operation mode: independent, tracking series, or tracking parallel.
+        NR1 0: Independent, 1: Series, 2: Parallel""",
+        validator=strict_discrete_set,
+        values=_TRACK_MODES,
```

Comment:
Consider using an enum or strings and then use a dict `{"independent": 0, ...}` and set `map_values=True`. This will map the strings to numbers and send the numbers to the device.